src/registry.c

```diff
@@ -870,11 +871,35 @@ parse_variant(struct rxkb_context *ctx, struct rxkb_layout *l,
                 if (!is_node(ci, "configItem"))
                     continue;
 
+                bool found_language_list = false;
+                bool found_country_list = false;
                 for (node = ci->children; node; node = node->next) {
-                    if (is_node(node, "languageList"))
+                    if (is_node(node, "languageList")) {
                         parse_language_list(node, v);
-                    if (is_node(node, "countryList"))
+                        found_language_list = true;
+                    }
+                    if (is_node(node, "countryList")) {
                         parse_country_list(node, v);
+                        found_country_list = true;
+                    }
+                }
+                if (!found_language_list) {
+                    // inherit from parent layout
+                    struct rxkb_iso639_code* x;
+                    list_for_each(x, &l->iso639s, base.link) {
+                        struct rxkb_iso639_code* code = rxkb_iso639_code_create(&v->base);
+                        code->code = strdup_safe(x->code);
```

Comment:
It does not make a lot of sense to me that `x->code` can be NULL - in such a case it should not exist in the first place. E.g. I don't expect users of `rxkb_iso639_code_get_code` to handle NULL (and it's not documented as being able to return NULL).

Were you able to figure out which file caused this?

Reply:
No. Not even sure where the "I/O warning : failed to load external entity" error is thrown.

https://github.com/xkbcommon/libxkbcommon/runs/4612955170?check_suite_focus=true

Reply:
https://github.com/xkbcommon/libxkbcommon/blob/master/src/registry.c

> libxml2 ... My enthusiasm about this is indescribable

Interestingly from the 'install dependencies' step https://github.com/xkbcommon/libxkbcommon/runs/4612955170?check_suite_focus=true

> libxml2 is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

Maybe macosx is using an older version of libxml2 which loads external entities by default: https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html#libxml2


> starting with libxml2 version 2.9, XXE has been disabled by default

Reply:
A suspicion: Is xmlCleanupParser called early?

Reply:
`"/tmp//rules/evdev.xml"` has  a double-slash suggests something that may have been a `$foo` is missing. and it starts with `/tmp` in the tool-options test, so it's likely related to the env vars in `test/tool-option-parsing.py:main`. Not 100% sure on the rest yet, this was just from a quick peek

Reply:
ok, had another look and that warning is a red herring, I think, and it doesn't even matter for this particular test. The stdout of the tool seems fine, stderr has that warning but that warning happens *before* the stdout so it's not the reason we abort.

I've filed #277 now to at least print the exit status to get a hint what's going on but I haven't been able to reproduce this myself in the CI. I'm half-wondering if it's just a side-effect of #275 though it really shouldn't be.

Reply:
`strdup` instead of `strdup_safe` works as you suggest.